src/context/ThemeContext.tsx

```diff
@@ -0,0 +1,32 @@
+import { createContext, useEffect, useState } from "react";
+
+type ThemeContextType = {
+  theme: string;
+  toggleTheme: () => void;
+};
+export const ThemeContext = createContext<ThemeContextType | undefined>(
+  undefined
+);
+
+export default function ThemeContextProvider({
```

Comment:
A common pattern when working with contexts is to create a custom hook to access the context value. This make using the context easier and we can eliminate the undesired values before accessing the context such as in the Auth context.
You can see this vide about this pattern:
https://youtu.be/I7dwJxGuGYQ?si=rQOrQqwrJ-PzFTmm